packages/sdk/src/chains/EVM/genericMessage.ts

```diff
@@ -0,0 +1,178 @@
+            (transfer as Transfer<GenericMessage>).details.executeContractAddress,
+            (transfer as Transfer<GenericMessage>).details.maxFee,
+            transfer.sender,
+            (transfer as Transfer<GenericMessage>).details.executionData,
+          ),
+        });
+      }
+      default:
+        throw new Error(`Unsupported fee handler type`);
+    }
+  }
+
+  /**
+   * Builds an unsigned transfer transaction.
+   * Should be executed after the approval transactions.
+   *
+   * @param transfer
+   * @param fee
+   * @returns unsigned transfer transaction
+   */
+  public async buildTransferTransaction(
+    transfer: Transfer<TransferType>,
+    fee: EvmFee,
+  ): Promise<PopulatedTransaction> {
+    const bridge = Bridge__factory.connect(this.config.getDomainConfig().bridge, this.provider);
+    switch (transfer.resource.type) {
+      case ResourceType.PERMISSIONLESS_GENERIC: {
+        return await genericMessageTransfer({
+          depositor: transfer.sender,
+          bridgeInstance: bridge,
+          domainId: transfer.to.id.toString(),
+          resourceId: transfer.resource.resourceId,
+          feeData: fee,
+          executeContractAddress: (transfer as Transfer<GenericMessage>).details
+            .executeContractAddress,
+          executeFunctionSignature: (transfer as Transfer<GenericMessage>).details
+            .executeFunctionSignature,
+          executionData: (transfer as Transfer<GenericMessage>).details.executionData,
+          maxFee: (transfer as Transfer<GenericMessage>).details.maxFee,
+        });
+      }
+      default:
+        throw new Error(`Resource type not supported by generic message transfer`);
+    }
+  }
+
+  /**
+   * Creates a Transfer<Fungible> object which can then be used to estimate transfer
+   * fees as well as create the necessary desposit and approval transactions.
+   *
+   * @category Bridge deposit
+   * @param {string} sourceAddress - The address of the sender
+   * @param {number} destinationChainId - The (Para)ChainId of the destination chain
+   * @param {string} destinationAddress - The address of the recipient on the destination chain
+   * @param {string} resourceId - The ID of the resource being transferred
+   * @param {string} amount - The amount of tokens to be transferred. The amount should be in the lowest denomination possible on the source chain. If the token on source chain is configured to use 12 decimals and the amount to be transferred is 1 ETH, then amount should be passed in as 1000000000000
+   * @returns {Transfer<Fungible>} - The populated transfer object
+   * @throws {Error} - Source domain not supported, Destination domain not supported, Resource not supported
+   */
+  public createGenericMessageTransfer(
+    sourceAddress: string,
+    destinationChainId: number,
+    resourceId: string,
+    executeContractAddress: string,
+    executeFunctionSignature: string,
```

Comment:
Does it make sense to have this as destination abi encoded methods and then data would be the raw params which we can encode?
Just thinking what will happen if we add Substrate XCM generic transfer option.
Also, what is the better UX for users.


Can you please check how other SDKs are doing this just so we are compliant.

Reply:
@mpetrun5 just so that I understand you correctly, are you proposing that we pass in the encoded call-data rather than the signature + encoded params?

Reply:
When thinking again about it. Best not to for now.